### PR TITLE
Fix/update comment

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
#176 

Solution
========
What I/we did to solve this problem
the workflow will on be triggered when:
1. a commit merged into `main`
2. a pull request has been opened/synchronized/reopened
Screenshot 2023-08-02 at 2 55 04 PM

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Push a commit on a branch that is not associated with a pull request---(the Analyze workflow will not be triggered)

### CODE REVIEW

**Comment by mogres** [comment]
This looks great! We could also skip the entire workflow (including the analysis part) if a PR does not exist. What do you think about moving the check further up in the `.yml` file? This will save us storage and processing time. Or do we want to keep the results on AWS as a test?

image

**Comment by rugeli** [comment]
> This looks great! We could also skip the entire workflow (including the analysis part) if a PR does not exist. What do you think about moving the check further up in the `.yml` file? This will save us storage and processing time. Or do we want to keep the results on AWS as a test?
> 
> image

Good suggestion! If we don't have a strong need for keeping results from non-PR commits in AWS, then I'll go for saving storage and processing time.

File: .github/workflows/analyze.yml [hunk 1]
```diff
@@ -3,8 +3,24 @@ name: Analyze
...
 jobs:
+    Check_pr:
+        runs-on: ubuntu-latest
+        outputs:
+            is_pr: ${{ steps.check.outputs.is_pr }}
+        steps:
+            - name: Check if PR
+              id: check
+              run: |
+                  if [[ "${{ github.event.pull_request }}" != "" ]]; then
```
Comment (mogres) [inline .github/workflows/analyze.yml:14]: Looks like `github.event.pull_request` is empty for this PR, which results in this statement evaluating to `False`. Maybe test it by checking what is the value stored in `github.event.pull_request`?

**Comment by mogres** [comment]
Ran the following tests:

1. Pushing a commit to a draft PR triggers the workflow
2. Pushing a commit to a non-PR branch does not trigger the workflow
3. Pushing a commit to an existing PR triggers the workflow
4. Pushing multiple commits updates the existing comment

This is expected behavior and we should be good to merge!

**Review by mogres — APPROVED** [review]

**Review by meganrm — APPROVED** [review]
Looks good, wait for #185 to be merged.

**Pull request merged**